### ISSUE TITLE
Ignore files for upload

### ISSUE THREAD

**Issue by ayushkovskiy** [neuromation/platform-client-python#1182]
Some files and folders such as `__pycache__`, vitual env directory, etc. should not be uploaded via cp. In some cases, this will significantly improve performance by not doing needless work. Moreover, it will improve the general UX as `neuro cp` will become more transparent.

The idea is to use `.neuroignore` file similar to `.gitignore`. Note, there's already an implementation of the parser: https://pypi.org/project/gitignore-parser/

**Comment by ayushkovskiy** [comment]
or `.nmignore` since we have `.nmrc`. But I prefer the full name `.neuroignore` since it is more clear

**Comment by mariyadavydova** [comment]
I want to add that using `.gitignore` instead of `.neuroignore` won't help in this case, because the excludes are not identical: we would like to ignore data in git, but not in neuro.

I have a question, though: `.gitignore` is a complex thing; for example, one may have several `.gitignore`s inside the project, and all of them are taken into account; are we going to support the identical behavior or will we simplify and use only one file?

**Comment by ayushkovskiy** [comment]
@dalazx has proposed that we could re-use existing `.gitignore` file

**Comment by mariyadavydova** [comment]
I commented this exact proposition in the previous comment. 

> I want to add that using .gitignore instead of .neuroignore won't help in this case, because the excludes are not identical: we would like to ignore data in git, but not in neuro.

**Comment by ayushkovskiy** [comment]
> I commented this exact proposition in the previous comment.

Yeah sorry for that I just wanted to remember this point being on the call.

> because the excludes are not identical: we would like to ignore data in git, but not in neuro.

Actually, as @tvoinarovskyi and @asvetlov have pointed out, they are not completely identical: `.gitignore` ignores datasets, but `.neuroignore` does not. Maybe we could have something like `.neuro-unignore` to unignore files ignored in `.gitignore` :joy: Though, this is total over-engineering.

**Comment by ayushkovskiy** [comment]
Same is needed for download: WandB, set up from Jupyter notebook, by default writes logs to `./wandb/run-...` files.

**Comment by ayushkovskiy** [comment]
As a workaround, we could follow WandB to use an env variable: `NEURO_IGNORE_GLOBS=__pycache__/,*.log,*.db`

see https://docs.wandb.com/library/python/save#ignoring-certain-files

**Comment by asvetlov** [comment]
Did you mean `WANDB_IGNORE_GLOBS`?

**Comment by ayushkovskiy** [comment]
yes, I meant `WANDB_IGNORE_GLOBS=*.patch` from the documentation examples.

**Comment by ayushkovskiy** [comment]
Note: AWS uses CLI options instead of a file, and I like it:

```
$ aws s3 help
...
      You can do
       this by providing an --exclude or --include  argument  multiple  times,
       e.g.   --include  "*.txt"  --include  "*.png".  When there are multiple
       filters, the rule is the filters that appear later in the command  take
       precedence  over filters that appear earlier in the command.  For exam-
       ple, if the filter parameters passed to the command were

          --exclude "*" --include "*.txt"
```
we could implement this instead.

**Comment by asvetlov** [comment]
We could even implement both, I see how every implementation can be useful.
The problem is the developer's time only.

**Comment by asvetlov** [comment]
Fixed by #1276
